packages/core/lib/src/widgets/tweet/retweets.dart

```diff
@@ -0,0 +1,55 @@
+import 'package:core/src/styles/app_colors.dart';
+import 'package:core/src/widgets/tweet/tweet_metric.dart';
+import 'package:flutter/material.dart';
+
+/// Tweet retweets icon and metric text widget
+class Retweets extends StatefulWidget {
+  /// Creates new instance of [Retweets]
+  const Retweets({
+    super.key,
+    this.metricValue = 0,
+    this.iconSize = 15,
+    this.onRetweet,
+    this.onQuoteTweet,
+  });
+
+  /// Retweets metric value
+  final int metricValue;
+
+  /// Size of retweets icon
+  final double iconSize;
+
+  /// Callback to notify parent widget of Retweet/Undo Retweet actions
+  final ValueChanged<bool>? onRetweet;
+
+  /// Callback to notify parent widget of quote tweet action
+  final VoidCallback? onQuoteTweet;
+
+  @override
+  State<Retweets> createState() => _RetweetsState();
+}
+
+class _RetweetsState extends State<Retweets> {
+  bool _isActive = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return TweetMetric(
+      isActive: _isActive,
+      metricValue: widget.metricValue,
+      // Todo: change to design icons
+      icon: const Icon(Icons.repeat),
+      activeIcon: const Icon(Icons.repeat),
+      iconSize: widget.iconSize,
+      activeColor: AppColors.success,
+      onPressed: () {
+        // Todo: This should open a dropdown of "Retweet/Quote Tweet"
+        // Only Retweet activates the metric
+        // Quote Tweet should open a dialog to submit a tweet
+        setState(() {
+          _isActive = !_isActive;
+        });
```

Comment:
We should create Issues if you have open Todo's merged into staging. GitHub has a syntax to parse these todos, which makes them easily accessible.

Reply:
Should we use this or do you have something else in mind?
https://github.com/marketplace/actions/todo-to-issue